bio_files_processor.py

```diff
@@ -0,0 +1,144 @@
+                        line = file.readline()
+                        if not line:
+                            break
+                    gene_info.append(info_line.rstrip())
+                line = file.readline()
+            line = file.readline()
+            genes.append(gene_info)
+    return genes
+
+
+def extract_genes(gbk_genes_info: list, inp_genes: list, n_before: int = 1, n_after: int = 1)->list:
+    """
+    Extract genes from gbk-list 
+    arguments:
+        - gbk_genes_info (list): information from gbk-format in type of list
+        - inp_genes (list): gene names, which are interested
+        - n_before (int): additional genes before the interesting one
+        - n_after (int): additional genes after the interesting one
+    return:
+        - list: extracted information from gbk-format about interesting and additional genes
+    """
+    genes_to_file = []
+    if not isinstance(inp_genes, list):
+        genes = [inp_genes]
+    for gene in genes:
+        gene = 'gene="' + gene + '"'
+        matching = [gbk_genes_info.index(gene_info) for gene_info in gbk_genes_info if gene in gene_info]
+    for matched_index in matching:
+        if matched_index-n_before <0:
+            start = 0
+        else: start = matched_index-n_before
+        if (matched_index+n_after+1) > len(gbk_genes_info):
+            end = len(gbk_genes_info)
+        else:
+            end = matched_index+n_after+1
+        for i in range (start, end):
+            genes_to_file.append(gbk_genes_info[i])
+    return genes_to_file
+
+
+def write_to_fasta(extracted_genes: list, output_fasta: str):
+    """
+    Write genes sequences in gbk-format to fasta-file
+    arguments:
+        - extracted_genes (list): information about genes in gbk-format in type of list
+        - output_fasta (str): name of fasta-file
+    """
+    with open(output_fasta, mode = 'w') as file:
+        for gene_info in extracted_genes:
+            name_string = '>'
+            for info in gene_info:
+                if not info.startswith('translation='):
+                    name_string = name_string + info + '|'
+                else:
+                    file.write(name_string[:-1]+'\n')
+                    file.write(info.split('translation="')[1]+'\n')
+
+
+def select_genes_from_gbk_to_fasta(input_gbk, genes, n_before = 1, n_after = 1, output_fasta = ''):
+    """
+    Extract the chosen genes from gbk-file and write them to fasta-file
+    arguments:
+        - input_gbk (str): path to the gbk-file for processing
+        - genes (list): genes names, which are interested
+        - n_before (int): additional genes before the interesting one
+        - n_after (int): additional genes after the interesting one
+        - output_fasta (str): name of fasta-file for saving the result
+    """
+    gbk_genes_info = read_gbk_to_list(input_gbk)
+    extracted_genes = extract_genes(gbk_genes_info, genes, n_before=1, n_after=1)

```

Comment:
При вызове функции select_genes_from_gbk_to_fasta все хорошо, параметры n_before и n_after имеют значение по умолчанию. Функция получает их в виде переменных с соответстующими названиями.
Но завтем, в вызове extract_genes снова используются аргументы по умолчанию, а не эти переменные. Это приводит к неправильной работе функции, если туда подаются аргументы, помимо 1.

PS PEP8, пробелы
Нет аннотации типов
```suggestion
def select_genes_from_gbk_to_fasta(input_gbk, genes, n_before=1, n_after=1, output_fasta=''):
    """
    Extract the chosen genes from gbk-file and write them to fasta-file
    arguments:
        - input_gbk (str): path to the gbk-file for processing
        - genes (list): genes names, which are interested
        - n_before (int): additional genes before the interesting one
        - n_after (int): additional genes after the interesting one
        - output_fasta (str): name of fasta-file for saving the result
    """
    gbk_genes_info = read_gbk_to_list(input_gbk)
    extracted_genes = extract_genes(gbk_genes_info, genes, n_before=1, n_after=1)

```